python/tchannel/tornado/connection.py

```diff
@@ -226,6 +224,10 @@ def set_close_callback(self, callback):
 
     def finish(self):
         """ write response """
-        self.response.arg_1 = "from inbound"
+        self.response.arg_1 = "response from inbound server"
         self.response.arg_2 = "inbound"
-        return self.frame_and_write(self.response)
+        self.frame_and_write(self.response)
+
+        # reset response message
+        self.response = CallResponseMessage()
```

Comment:
Why is all this being tracked on the same instance? Aren't we expecting the same TornadoConnection to handle multiple possibly out-of-order requests?